src/vs/workbench/api/node/extHostSearch.ts

```diff
@@ -35,17 +36,24 @@ export class NativeExtHostSearch extends ExtHostSearch implements IDisposable {
 		@IExtHostRpcService extHostRpc: IExtHostRpcService,
 		@IExtHostInitDataService initData: IExtHostInitDataService,
 		@IURITransformerService _uriTransformer: IURITransformerService,
+		@IExtHostConfiguration private readonly configurationService: IExtHostConfiguration,
 		@ILogService _logService: ILogService,
 	) {
 		super(extHostRpc, _uriTransformer, _logService);
-
+		this.getNumThreads = this.getNumThreads.bind(this);
 		const outputChannel = new OutputChannel('RipgrepSearchUD', this._logService);
-		this._disposables.add(this.registerTextSearchProvider(Schemas.vscodeUserData, new RipgrepSearchProvider(outputChannel)));
+		this._disposables.add(this.registerTextSearchProvider(Schemas.vscodeUserData, new RipgrepSearchProvider(outputChannel, this.getNumThreads)));
 		if (initData.remote.isRemote && initData.remote.authority) {
 			this._registerEHSearchProviders();
 		}
 	}
 
+	async getNumThreads(): Promise<number | undefined> {
+		const configProvider = await this.configurationService.getConfigProvider();
```

Comment:
Rather than querying the configuration service for every time we search, I think that using storing the `numThreads` as a property and using the configProvider once to listen to `onDidChangeConfiguration` and using `affectsConfiguration` on the event would be more efficient.